src/Split.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-
+
+The high-level idea behind this implementation:
+
+* Given an original word, like "cart"
+* And a list of rhymes, like "heart" and "start"
+* And some phrases, like "put your heart in it" and "kick start"
+* We want to create "put your cart into it" and "kick cart"
```

Comment:
So...

```hs
import Data.Text (Text)
import qualified Data.Text as T

replaceWord :: Text -> Text -> Text -> Text
replaceWord from to = T.unwords . map swap . T.words

  where
    swap x
        | x == from = to
        | otherwise = x
```

?